### PR TITLE
devp2p base Golem service

### PR DESCRIPTION
**Pull request by Elfoniok**
Signed-off-by: Lukasz Foniok <lukaszfoniok@gmail.com>

### CODE REVIEW

File: golem/client.py [hunk 1]
```diff
@@ -523,29 +494,28 @@ def get_node_name(self):
         return str(name) if name else ''
 
     def get_neighbours_degree(self):
-        return self.p2pservice.get_peers_degree()
+        pass
```
Comment (badb) [inline golem/client.py]: Can you add TODO comment or an issue about reimplementing this method with devp2p?

File: golem/client.py [hunk 2]
```diff
@@ -523,29 +494,28 @@ def get_node_name(self):
...
-        return self.p2pservice.suggested_address.get(key_id)
+        return self.services['golem_service'].suggested_address.get(key_id)
 
     def get_suggested_conn_reverse(self, key_id):
-        return self.p2pservice.get_suggested_conn_reverse(key_id)
+        return self.services['golem_service'].get_suggested_conn_reverse(key_id)
 
     def get_resource_peers(self):
-        self.p2pservice.send_get_resource_peers()
+        pass
```
Comment (badb) [inline golem/client.py]: Do we want to remove this method completely? I assume that it won't be very useful for future  resource sending mechanism.

File: golem/client.py [hunk 3]
```diff
@@ -715,7 +682,7 @@ def use_ranking(self):
         return bool(self.ranking)
 
     def want_to_start_task_session(self, key_id, node_id, conn_id):
-        self.p2pservice.want_to_start_task_session(key_id, node_id, conn_id)
+        pass
```
Comment (badb) [inline golem/client.py]: Can you add TODO comment or throw NotImplemented error here? Same with the functions below.

File: golem/client.py (hunk 2 above)
Comment (Elfoniok) [reply golem/client.py]: We should remove this, for long time there is no such thing as resource peer.

File: golemapp.py [hunk 4]
```diff
@@ -1,21 +1,35 @@
 #!/usr/bin/env python
+import os
 import sys
+# Set the default event loop
+os.environ['GEVENT_LOOP'] = 'tulipcore.Loop'
```
Comment (badb) [inline golemapp.py:5]: Dummy task fails if it's run separetly from full test suite. This variable is not set there.

File: golem/client.py [hunk 5]
```diff
@@ -1051,8 +1027,9 @@ def __publish_events(self):
                 })
 
     def __make_node_state_snapshot(self, is_running=True):
-        peers_num = len(self.p2pservice.peers)
-        last_network_messages = self.p2pservice.get_last_messages()
+        peers_num = 0  # len(self.p2pservice.peers)
+        # TODO: Is it necessary to collect and send network messages?
```
Comment (badb) [inline golem/client.py:1037]: https://github.com/golemfactory/golem/issues/1424 That's for the future.

File: golem/p2pconfig.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+from golem.core.variables import APP_NAME, APP_VERSION
```
Comment (badb) [inline golem/p2pconfig.py:1]: https://github.com/golemfactory/golem/issues/1425 <- Issue for the future

**Comment by badb** [comment]
```
INFO:p2p.peer	received hello version=55 client_version=b'Brass Golem 0.9.0' capabilities=((b'p2p', 4), (b'golem_proto', 1), (b'task_proto', 1))
[]
Traceback (most recent call last):
  File "C:\Users\badb\AppData\Local\Programs\Python\Python36\lib\site-packages\gevent\greenlet.py", line 536, in run
    result = self._run(*self.args, **self.kwargs)
  File "C:\Users\badb\AppData\Local\Programs\Python\Python36\lib\site-packages\devp2p\peer.py", line 232, in _run_decoded_packets
    self._handle_packet(self.mux.packet_queue.get())  # get_packet blocks
  File "C:\Users\badb\AppData\Local\Programs\Python\Python36\lib\site-packages\devp2p\peer.py", line 206, in _handle_packet
    protocol.receive_packet(packet)
  File "C:\Users\badb\AppData\Local\Programs\Python\Python36\lib\site-packages\devp2p\protocol.py", line 165, in receive_packet
    cmd(packet)
  File "C:\Users\badb\AppData\Local\Programs\Python\Python36\lib\site-packages\devp2p\protocol.py", line 137, in receive
    instance.receive(proto=self, data=klass.decode_payload(packet.payload))
  File "C:\Sources\golem\golem\network\p2p\protocol.py", line 80, in decode_payload
    return signed.decode_payload(rlp_data)
  File "C:\Users\badb\AppData\Local\Programs\Python\Python36\lib\site-packages\devp2p\protocol.py", line 103, in decode_payload
    raise e
  File "C:\Users\badb\AppData\Local\Programs\Python\Python36\lib\site-packages\devp2p\protocol.py", line 100, in decode_payload
    data = rlp.decode(rlp_data, sedes=decoder)
  File "C:\Users\badb\AppData\Local\Programs\Python\Python36\lib\site-packages\rlp\codec.py", line 216, in decode
    obj = sedes.deserialize(item, **kwargs)
  File "C:\Users\badb\AppData\Local\Programs\Python\Python36\lib\site-packages\rlp\sedes\lists.py", line 88, in deserialize
    raise ListDeserializationError('List has wrong length', serial)
rlp.exceptions.ListDeserializationError: List has wrong length
``` 
Is it a bug or a feature?

File: requirements.txt [hunk 7]
```diff
@@ -41,15 +41,18 @@ qt5reactor
 GitPython>=2.1.0
 semantic_version
 pyOpenSSL==16.2.0
+gevent==1.2.2
+devp2p==0.9.2
+rlp==0.5.1
```
Comment (badb) [inline requirements.txt:46]: Is there  a reason to use rlp==0.5.1 and not 0.6.0?

File: golemapp.py (hunk 4 above)
Comment (mfranciszkiewicz) [reply golemapp.py:5]: Fixed in #1426

**Comment by mfranciszkiewicz** [comment]
@badb The bootstrap node was not updated with the latest changes.

File: golem/network/p2p/protocol.py [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+        @classmethod
+        def decode_payload(cls, rlp_data):
+            return signed.decode_payload(rlp_data)
+
+        @classmethod
+        def verify_message(cls, proto, payload, ts, sig):
+            payload_digest = digest(payload, ts)
+            if not proto.verify(sig, payload_digest):
+                raise AssertionError('Invalid message signature')
```
Comment (jiivan) [inline golem/network/p2p/protocol.py:86]: I suggest refraining from using `AssertionError` here. It generally should be used with `assert` statement to indicate a condition that should never occur. Also `AssertionError`s are disabled when python runs in optimized mode (see: https://docs.python.org/3/library/constants.html#__debug__ ).
Maybe defining custom `ProtocolError` or just using generic `RuntimeError` will do?

**Review by jiivan — COMMENTED** [review]
Overall I really like refactorings made within this PR. It's much cleaner and easier to read.

**Pull request closed without merging**